### PR TITLE
tidy up checkVarNames()

### PR DESCRIPTION
**Pull request by beckerbenj**
Helper function `transf_names()` was outdated, slow and not vectorized. This PR bundles all functionality in the S3 generic `checkVarNames()`. Documentation is extended and all use cases of `checkVarNames()` (formerly `transf_names()`) are revised. 

This prepares a planned fix for issue #46 .

### CODE REVIEW

File: R/getChangeMeta.R [hunk 1]
```diff
@@ -70,11 +70,8 @@ check_varChanges <- function(changeTable) {
...
   # tbd: content checks for format and display width
   # SQLite compliance
-  changeTable$varName_new <- sapply(changeTable$varName_new, function(x) {
-    if(is.na(x)) return(NA)
-    transf_names(x)
-    })
-
+  not_na <- !is.na(changeTable$varName_new)
+  changeTable$varName_new[not_na] <- checkVarNames(as.character(changeTable$varName_new[not_na]))
   changeTable
```
Comment (franikowsp) [inline R/getChangeMeta.R:75]: LGTM.

File: R/prepare_labels.R [hunk 2]
```diff
@@ -42,20 +42,6 @@ unduplicate <- function(x) {
...
-  NewName <- vec_name
-  if(any(grepl(paste0("^", vec_name, "$"), eatDB::sqlite_keywords, ignore.case = TRUE))) {
-    NewName <- paste0(vec_name, "Var")
-  }
-  NewName <- make.names(NewName)
-  if(grepl("\\.", NewName))       NewName <- gsub("\\.", "_", NewName)
-
-  if(!identical(NewName, vec_name)) message(paste(vec_name, "has been renamed to", NewName))
-  NewName
-}
```
Comment (franikowsp) [inline R/prepare_labels.R:57]: I think, the new code is way clearer and more capable now! 🎉

**Review by franikowsp — COMMENTED** [review]
Only very tiny nits (and one potential typo) - the rest looks very good to me! Thank you for providing the context (otherwise, this would have been a little bit too abstract to review).

File: R/checkVarNames.R [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
-#' Check names for \code{SQLite} conventions.
+#' Check names for \code{SQLite} column name conventions.
 #'
-#' Applies variable names changes to \code{GADSdat} or \code{all_GADSdat} objects.
+#' Checks names for \code{SQLite} column name conventions and
+#' applies appropriate variable name changes to \code{GADSdat} or \code{all_GADSdat} objects.
 #'
-#' Illegal names in a \code{SQLite} data base include \code{SQLite} keywords (see \code{\link[eatDB]{sqlite_keywords}}) and names
-#' with a \code{"."} in it.
+#' Illegal column names in a \code{SQLite} data base include
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply R/checkVarNames.R:9]: 😆 Agree and fixed!

File: R/checkVarNames.R [hunk 4]
```diff
@@ -20,24 +35,56 @@
...
+checkVarNames.data.frame <- function(GADSdat, checkKeywords = TRUE, checkDots = TRUE) {
+  names(GADSdat) <- checkVarNames(names(GADSdat))
+  GADSdat
+}
+#'@export
+checkVarNames.character <- function(GADSdat, checkKeywords = TRUE, checkDots = TRUE) {
+  NewName <- GADSdat
+  check_logicalArgument(checkKeywords)
+  check_logicalArgument(checkDots)
+  if(any(is.na(GADSdat))) stop("Column names can not be NA.")
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply R/checkVarNames.R:68]: (Kind of 😛 ) agree and fixed.

File: tests/testthat/test_checkVarNames.R [hunk 5]
```diff
@@ -14,6 +18,23 @@ df4$labels[3, "missings"] <- "valid"
 df4$labels[2:3, "labeled"] <- "yes"
 
 
+test_that("Variable name are transformed correctly", {
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply tests/testthat/test_checkVarNames.R:21]: Seperated those tests and named them as suggested.

**Pull request merged**